src/eventListeners/colony.ts

```diff
@@ -96,5 +108,23 @@ export const setupListenersForColony = (colonyAddress: string): void => {
     addColonyEventListener(eventSignature, colonyAddress),
   );
 
-  addTokenEventListener(ContractEventsSignatures.Transfer, colonyAddress);
+  const tokenEvents = [
+    ContractEventsSignatures.Transfer,
+    ContractEventsSignatures.LogSetAuthority,
+  ];
+
+  /*
+   * @TODO once there are more relevant events in this list, we should extract this
+   * listener separately and run it for each token we see within our network
+   *
+   * Currently it works as we only care about one specific event which only happens
+   * on colony creation (Transfer doesn't count as it's being handled differently)
+   */
+  tokenEvents.forEach((eventSignature) => {
+    // This needs to be handled individually because transfer events are frequent and heavy
+    if (eventSignature === ContractEventsSignatures.Transfer) {
+      return addTokenTransferEventListener(eventSignature, colonyAddress);
+    }
+    return addTokenEventListener(eventSignature, tokenAddress);
```

Comment:
As discussed on a call, currently this potentially has quite poor performance if multiple colonies are using the same token and a `LogSetAuthority` event is fired - there will be multiple handlers - one created for each colony - and each call with update the appropriate values of all relevant colonies.

As further discussed on the call, using our [existing list of handlers](https://github.com/JoinColony/block-ingestor/blob/master/src/eventListeners/eventListeners.ts#L5) might be a quick way to improve this, checking if we have a relevant handler already.